File: 04-project-loom/src/main/java/pro/seag/controller/ApplicationController.java
package pro.seag.controller;

import lombok.AllArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.client.RestTemplate;

@RestController
@RequestMapping
@AllArgsConstructor
public class ApplicationController {

    private final RestTemplate restTemplate;

    @GetMapping
    public ResponseEntity<Object> callSlowServer(){
        return restTemplate.getForEntity("http://127.0.0.1:8000", Object.class);
    }

}

File: 03-coroutines/src/main/kotlin/pro/seag/controller/ApplicationController.kt
package pro.seag.controller

import kotlinx.coroutines.flow.Flow
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.awaitExchange
import org.springframework.web.reactive.function.client.bodyToFlow

@RestController
@RequestMapping
open class ApplicationController(private val slowServerClient: WebClient) {

    @GetMapping
    suspend fun callSlowServer(): Flow<Any> {
        return slowServerClient.get().awaitExchange().bodyToFlow(String::class)
    }

}